demos/smart_framing_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -0,0 +1,287 @@
+            even_rect.width = (init_rect.width % 2 == 0) ? (init_rect.width) : (init_rect.width - 1);
+            even_rect.height = (init_rect.height % 2 == 0) ? (init_rect.height) : (init_rect.height - 1);
+
+            cv::Mat SF_ROI;
+            image(even_rect).copyTo(SF_ROI);
+
+            cv::Mat SF_resized_ROI;
+            cv::Size target_size;
+            target_size.height = image.size().height;
+            target_size.width = even_rect.width * (static_cast<float>(image.size().height) / static_cast<float>(even_rect.height));
+            target_size.width = (target_size.width % 2 == 0) ? (target_size.width) : (target_size.width - 1);
+            if (target_size.width > image.size().width) {
+                target_size.width = image.size().width;
+            }
+            cv::resize(SF_ROI, SF_resized_ROI, target_size);
+            int left_add = (image.size().width - target_size.width) / 2;
+            int right_add = (image.size().width - target_size.width) / 2;
+            cv::copyMakeBorder(/*croped_rect*/ SF_resized_ROI, out,
+                0,
+                0,
+                left_add,
+                right_add,
+                cv::BORDER_CONSTANT, cv::Scalar::all(127));
+        }
+        else {
+            image.copyTo(out);
+        }
+    }
+};
+
+GAPI_OCV_KERNEL(OCVSuperResolutionPostProcessingKernel, custom::GSuperResolutionPostProcessingKernel) {
+    static void run(const cv::Mat & image, cv::Mat & out) {
+        float* outputData = (float*)image.data;
+        size_t outChannels = image.size[1];
+        size_t outHeight = image.size[2];
+        size_t outWidth = image.size[3];
+        size_t numOfPixels = outWidth * outHeight;
+        std::vector<cv::Mat> imgPlanes;
+        if (outChannels == 3) {
+            imgPlanes = std::vector<cv::Mat>{
+                  cv::Mat(outHeight, outWidth, CV_32FC1, &(outputData[0])),
+                  cv::Mat(outHeight, outWidth, CV_32FC1, &(outputData[numOfPixels])),
+                  cv::Mat(outHeight, outWidth, CV_32FC1, &(outputData[numOfPixels * 2])) };
+        }
+        else {
+            imgPlanes = std::vector<cv::Mat>{ cv::Mat(outHeight, outWidth, CV_32FC1, &(outputData[0])) };
+            // Post-processing for text-image-super-resolution models
+            cv::threshold(imgPlanes[0], imgPlanes[0], 0.5f, 1.0f, cv::THRESH_BINARY);
+        }
+        for (auto& img : imgPlanes)
+            img.convertTo(img, CV_8UC1, 255);
+
+        cv::merge(imgPlanes, out);
+    }
+};
+
+
+// NB: cv::gapi::convertTo + reshape from 4D blob to image
+GAPI_OCV_KERNEL(OCVCvt32Fto8U, custom::GCvt32Fto8U) {
+    static void run(const cv::Mat & in,
+        cv::Mat & out) {
+        int h = in.size[2];
+        int w = in.size[3];
+
+        auto* out_p = out.ptr<uint8_t>();
+        auto* in_p = in.ptr<const float>();
+
+        std::transform(in_p, in_p + h * w, out_p,
+            [](float v) { return static_cast<uint8_t>(v * 255); });
+    }
```

Comment:
Can it be:
out = in * 255;
Are in and out 1-channel cv::Mat?

Reply:
Need to check with Tolik. This is his kernel.